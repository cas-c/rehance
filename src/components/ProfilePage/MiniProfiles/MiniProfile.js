import React from 'react';
import PropTypes from 'prop-types';

const MiniProfile = ({ user }) => (
    <div className='card mini-profile'>
        <div className='card-content'>
            <div className='media'>
                <div className='media-left'>
                    <figure className='image is-48x48'>
                        <img alt='' src={user.images['115']} />
                    </figure>
                </div>
                <div className='media-content'>
                    <p className='title is-5'>{user.display_name}</p>
                    <p className='subtitle is-6'>{user.username}</p>
                </div>
            </div>
        </div>
    </div>
);

MiniProfile.propTypes = {
    user: PropTypes.shape({
        images: PropTypes.objectOf(PropTypes.string).isRequired,
        display_name: PropTypes.string,
        username: PropTypes.string
    })
}

MiniProfile.defaultProps = {
    user: {
        images: { 115: 'placeholder image url' },
        display_name: 'Default User',
        username: 'username'
    }
}

export default MiniProfile;
