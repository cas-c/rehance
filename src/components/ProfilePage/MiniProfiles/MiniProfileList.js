import React from 'react';
import PropTypes from 'prop-types';
import MiniProfile from './MiniProfile';

const MiniProfiles = ({ users }) => users.slice(0, 5).map((u, i) => <MiniProfile user={u} key={`mini-user-${i}`} />);

MiniProfiles.propTypes = {
    users: PropTypes.arrayOf(PropTypes.object)
};

MiniProfiles.defaultProps = {
    users: []
}

export default MiniProfiles;
