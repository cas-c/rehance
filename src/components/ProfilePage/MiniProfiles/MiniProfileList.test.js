import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import MiniProfileList from './MiniProfileList';
import MiniProfile from './MiniProfile';

const mockUsersLessThanFive = [
    {
        images: {
            115: 'testImageUrl'
        },
        display_name: 'Test User',
        username: 'miniprofileuser'
    },
    {
        images: {
            115: 'testImageUrl2'
        },
        display_name: 'Test User 2',
        username: 'miniprofileuser2'
    }
];

const mockUsersMoreThanFive = [
    {
        images: {
            115: 'testImageUrl'
        },
        display_name: 'Test User',
        username: 'miniprofileuser'
    },
    {
        images: {
            115: 'testImageUrl2'
        },
        display_name: 'Test User 2',
        username: 'miniprofileuser2'
    },
    {
        images: {
            115: 'testImageUrl3'
        },
        display_name: 'Test User 3',
        username: 'miniprofileuser3'
    },
    {
        images: {
            115: 'testImageUrl4'
        },
        display_name: 'Test User 4',
        username: 'miniprofileuser4'
    },
    {
        images: {
            115: 'testImageUrl5'
        },
        display_name: 'Test User 5',
        username: 'miniprofileuser5'
    },
    {
        images: {
            115: 'testImageUrl6'
        },
        display_name: 'Test User 6',
        username: 'miniprofileuser6'
    },
    {
        images: {
            115: 'testImageUrl7'
        },
        display_name: 'Test User 7',
        username: 'miniprofileuser7'
    }
];

describe('MiniProfileList', () => {
    it('snapshot: should render correctly with less than five users', () => {
        const output = shallow(<MiniProfileList users={mockUsersLessThanFive}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('snapshot: should render correctly with more than five users', () => {
        const output = shallow(<MiniProfileList users={mockUsersMoreThanFive}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('snapshot: should not error out without input', () => {
        const output = shallow(<MiniProfile />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('unit: displays the proper amount', () => {
        const mounted = mount(<MiniProfileList users={mockUsersLessThanFive} />);
        expect(mounted.children().length).toBe(2);
        mounted.unmount();
        const mountedOverFive = mount(<MiniProfileList users={mockUsersMoreThanFive} />);
        expect(mountedOverFive.children().length).toBe(5);
        mountedOverFive.unmount();
    })
})