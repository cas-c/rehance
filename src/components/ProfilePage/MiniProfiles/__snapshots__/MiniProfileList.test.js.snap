// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniProfileList snapshot: should not error out without input 1`] = `
<div
  className="card mini-profile"
>
  <div
    className="card-content"
  >
    <div
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            alt=""
            src="placeholder image url"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <p
          className="title is-5"
        >
          Default User
        </p>
        <p
          className="subtitle is-6"
        >
          username
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`MiniProfileList snapshot: should render correctly with less than five users 1`] = `
Array [
  <MiniProfile
    key="mini-user-0"
    user={
        Object {
            "display_name": "Test User",
            "images": Object {
              "115": "testImageUrl",
            },
            "username": "miniprofileuser",
          }
    }
/>,
  <MiniProfile
    key="mini-user-1"
    user={
        Object {
            "display_name": "Test User 2",
            "images": Object {
              "115": "testImageUrl2",
            },
            "username": "miniprofileuser2",
          }
    }
/>,
]
`;

exports[`MiniProfileList snapshot: should render correctly with more than five users 1`] = `
Array [
  <MiniProfile
    key="mini-user-0"
    user={
        Object {
            "display_name": "Test User",
            "images": Object {
              "115": "testImageUrl",
            },
            "username": "miniprofileuser",
          }
    }
/>,
  <MiniProfile
    key="mini-user-1"
    user={
        Object {
            "display_name": "Test User 2",
            "images": Object {
              "115": "testImageUrl2",
            },
            "username": "miniprofileuser2",
          }
    }
/>,
  <MiniProfile
    key="mini-user-2"
    user={
        Object {
            "display_name": "Test User 3",
            "images": Object {
              "115": "testImageUrl3",
            },
            "username": "miniprofileuser3",
          }
    }
/>,
  <MiniProfile
    key="mini-user-3"
    user={
        Object {
            "display_name": "Test User 4",
            "images": Object {
              "115": "testImageUrl4",
            },
            "username": "miniprofileuser4",
          }
    }
/>,
  <MiniProfile
    key="mini-user-4"
    user={
        Object {
            "display_name": "Test User 5",
            "images": Object {
              "115": "testImageUrl5",
            },
            "username": "miniprofileuser5",
          }
    }
/>,
]
`;
